modules/pam_xauth/pam_xauth.c

```diff
@@ -99,9 +100,17 @@ run_coprocess(pam_handle_t *pamh, const char *input, char **output,
 	char *buffer = NULL;
 	size_t buffer_size = 0;
 	va_list ap;
+    struct sigaction newsa, oldsa;
 
 	*output = NULL;
 
+    memset(&newsa, '\0', sizeof(newsa));
+    newsa.sa_handler = SIG_DFL;
+    if (sigaction(SIGCHLD, &newsa, &oldsa) == -1) {
+        pam_syslog(pamh, LOG_ERR, "Cannot set signal value");
+        return PAM_SYSTEM_ERR;
+	}
+
```

Comment:
When indenting new code, please follow the indentation style of the surrounding code.